code/game/objects/micro_structures.dm

```diff
@@ -0,0 +1,201 @@
+				choice = tgui_input_list(user, "Where would you like to go?", "Pick a tunnel", destinations)
+				if(!choice)
+					return
+				to_chat(user,"<span class = 'notice'>You begin moving...</span>")
+				if(!do_after(user, 10 SECONDS, exclusive = TRUE))
+					return
+				user.forceMove(choice)
+				var/obj/structure/micro_tunnel/da_oddawun = choice
+				da_oddawun.tunnel_notify(user)
+				return
+
+	if(!can_enter(user))
+		user.visible_message("<span class = 'warning'>\The [user] reaches into \the [src]. . .</span>","<span class = 'warning'>You reach into \the [src]. . .</span>")
+		if(!do_after(user, 3 SECONDS, exclusive = TRUE))
+			user.visible_message("<span class = 'notice'>\The [user] pulls their hand out of \the [src].</span>","<span class = 'warning'>You pull your hand out of \the [src]</span>")
+			return
+		if(!src.contents.len)
+			to_chat(user, "<span class = 'warning'>There was nothing inside.</span>")
+			user.visible_message("<span class = 'notice'>\The [user] pulls their hand out of \the [src].</span>","<span class = 'warning'>You pull your hand out of \the [src]</span>")
+			return
+		var/grabbed = pick(src.contents)
+		if(!grabbed)
+			to_chat(user, "<span class = 'warning'>There was nothing inside.</span>")
+			user.visible_message("<span class = 'notice'>\The [user] pulls their hand out of \the [src].</span>","<span class = 'warning'>You pull your hand out of \the [src]</span>")
+			return
+
+		if(ishuman(user))
+			var/mob/living/carbon/human/h = user
+			var/mob/living/l = grabbed
+			if(isliving(grabbed))
+				l.attempt_to_scoop(h)
+			else
+				var/atom/movable/whatever = grabbed
+				whatever.forceMove(get_turf(src.loc))
+
+			user.visible_message("<span class = 'warning'>\The [user] pulls \the [grabbed] out of \the [src]! ! !</span>")
+			return
+
+		else if(isanimal(user))
+			var/mob/living/simple_mob/a = user
+			var/mob/living/l = grabbed
+			if(!a.has_hands || isliving(grabbed))
+				l.attempt_to_scoop(user)
+			else
+				var/atom/movable/whatever = grabbed
+				whatever.forceMove(get_turf(src.loc))
+			user.visible_message("<span class = 'warning'>\The [user] pulls \the [grabbed] out of \the [src]! ! !</span>")
+			return
+
+	if(tgui_alert(user,"Do you want to go into the tunnel?","Enter Tunnel",list("Yes", "No")) != "Yes")
+		return
+	user.visible_message("<span class = 'notice'>\The [user] begins climbing into \the [src]!</span>")
+	if(!do_after(user, 10 SECONDS, exclusive = TRUE))
+		to_chat(user, "<span class = 'warning'>You didn't go into \the [src]!</span>")
+		return
+
+	enter_tunnel(user)
+
+/obj/structure/micro_tunnel/proc/can_enter(var/mob/living/user)
+	if(user.mob_size <= MOB_TINY || user.size_multiplier <= max_accepted_scale)
+		return TRUE
+
+	return FALSE
+
+/obj/structure/micro_tunnel/attack_generic(mob/user, damage, attack_verb)
+	attack_hand(user)
+	return ..()
+
+/obj/structure/micro_tunnel/MouseDrop_T(mob/living/M, mob/living/user)
+	. = ..()
+	if(M != user)
+		return
+
+	if(!can_enter(user))
```

Comment:
Maybe error message on trying to enter when too lorg?

Reply:
When you are too big to enter you put your hand in to try to pull things out!

Reply:
I mean specifically for interaction I commented on, where you click-n-drag yourself onto the hole